hal_st/middlewares/ble_middleware/GapCentralSt.cpp

```diff
@@ -105,29 +106,10 @@ namespace hal
             connectionParameters.slaveLatency, connectionParameters.supervisorTimeoutMs);
     }
 
-    void GapCentralSt::SetPhy()
-    {
-        if (hci_le_set_phy(connectionContext.connectionHandle, allPhys, speed1Mbps | speed2Mbps, speed1Mbps | speed2Mbps, 0) != BLE_STATUS_SUCCESS)
-            infra::EventDispatcherWithWeakPtr::Instance().Schedule([this]() { this->SetPhy(); });
-        else
-            infra::EventDispatcherWithWeakPtr::Instance().Schedule([this]() { this->DataLengthUpdate(); });
-    }
-
     void GapCentralSt::DataLengthUpdate()
     {
         if (hci_le_set_data_length(connectionContext.connectionHandle, transmissionOctets, transmissionTime) != BLE_STATUS_SUCCESS)
```

Comment:
Why do we block statechanged notification if we cannot do a data length update?
We are still connected. I would separate the two flows

Reply:
When DataLengthUpdate is called, a connection is already configured, which means that we can do a data length update.

By separating this in two flows you mean one function to set the connection e another to set the data length, but who is going to be the responsible to handle that? Application?